sound/soc/sof/ipc4.c

```diff
@@ -892,6 +892,15 @@ void sof_ipc4_mic_privacy_state_change(struct snd_sof_dev *sdev, bool state)
 	struct sof_ipc4_msg msg;
 	u32 data = state;
 
+	/*
+	 * If the firmware has not been booted up, there is no need to send
+	 * change notification. The firmware only needs to be aware of a change,
+	 * when it boots up the privacy state is checked by the firmware, it
+	 * only needs to be notified if it has changed
+	 */
+	if (sdev->fw_state != SOF_FW_BOOT_COMPLETE)
+		return 0;
+
```

Comment:
Will firmware check the privacy state when it boots up?

Reply:
and if the firmware is not booted up then it does not need to be notified of a change (as it is not booted up)